# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/device/gl.rs
# Commit: 0edb6b6f17b2
# Full Hash: 0edb6b6f17b24c737d053df19730e9c9e61e17ca
# Author: Glenn Watson <git@intuitionlibrary.com>
# Date: 2020-12-18 04:07:40
# Regressor Bug: 1682365
# File Overlap Count: 1
# Description:
#   Bug 1682365 - Pt 2 - Remove texture array usage from render task graph. r=nical
#   
#   With this change, all color/alpha intermediate surfaces are individual
#   textures, rather than texture arrays.
#   
# ==============================================================================

diff -r ee0de0eac18d -r 0edb6b6f17b2 gfx/wr/webrender/src/device/gl.rs
--- a/gfx/wr/webrender/src/device/gl.rs	Thu Dec 17 22:26:08 2020 +0000
+++ b/gfx/wr/webrender/src/device/gl.rs	Thu Dec 17 22:26:10 2020 +0000
@@ -475,6 +475,13 @@
         self.filter
     }
 
+    pub fn is_array(&self) -> bool {
+        match self.target {
+            gl::TEXTURE_2D_ARRAY => true,
+            _ => false
+        }
+    }
+
     pub fn supports_depth(&self) -> bool {
         !self.fbos_with_depth.is_empty()
     }